content/en/docs/ops/configuration/traffic-management/traffic-routing/index.md

```diff
@@ -0,0 +1,118 @@
+
+If traffic cannot be matched using one of the methods described above, it is treated as [passthrough traffic](/docs/tasks/traffic-management/egress/egress-control/#envoy-passthrough-to-external-services).
+By default, these requests will be forwarded as-ism, which ensures that traffic to services that Istio is not aware of, such as external services that do not have `ServiceEntry`s created.
+Note that when these requests are forwarded, mutual TLS will not be used and telemetry collection is limited.
+
+## Service types
+
+Along with standard ClusterIP Services, Istio supports the full range of Kubernetes Services, with some caveats.
+
+### LoadBalancer and NodePort Services
+
+These Services are supersets of ClusterIP Services, and are mostly concerned with allow access from external clients.
+These service types are supported and behave exactly like standard ClusterIP Services.
+
+### Headless Services
+
+A [headless Service](https://kubernetes.io/docs/concepts/services-networking/service/#headless-services) is a Service that does not have a ClusterIP assigned.
+Instead, the DNS response will contain the IP addresses of each endpoint (ie the Pod IP) that is a part of the Service.
+
+In general, Istio does not configure listeners for each Pod IP, as it works at the Service level.
+However, to support headless services, listeners are set up for each IP:Port pair in the headless service.
+
+{{< warning >}}
+Without Istio, the `ports` field of a headless service is not strictly required because requests go directly to pod IPs, which can accept traffic on all ports.
+However, with Istio the port must be declared in the Service, or it will [not be matched](#unmatched-traffic).
+{{< /warning >}}
+
+### ExternalName Services
+
+An [ExternalName Service](https://kubernetes.io/docs/concepts/services-networking/service/#externalname) is essentially just a DNS alias.
+
+Because there is no ClusterIP nor pod IPs to match on, for TCP ExternalName Services, all IPs on the port will be matched.
+This may prevent [unmatched traffic](#unmatched-traffic) on the same port from being forwarded correctly.
+As such, it is best to avoid these where possible, or use dedicated ports when needed.
+HTTP and TLS do not share this constraint, as routing is done based on the hostname/SNI.
+
+{{< warning >}}
+Without Istio, the `ports` field of an ExternalName service is not required because the Service only represents a DNS entry.
+However, with Istio the port must be declared in the Service, or it will [not be matched](#unmatched-traffic).
+{{< /warning >}}
+
+### ServiceEntry
+
+In addition to Kubernetes Services, [ServiceEntries](/reference/config/networking/service-entry/#ServiceEntry) can be created to extend the set of services known to Istio.
+This can be useful to ensure that traffic to external services, such as `example.com`, get the functionality of Istio.
+
+A ServiceEntry with `addresses` set will perform routing just like a ClusterIP Service.
```

Comment:
I kind of get what this means but can you elaborate? If a SE has protocol TCP, then it needs to specify address for VIP matching else only ports are matched. Does resolution:DNS work in that case (I think it should) or users need to provide static IP addresses as endpoints?

Reply:
I will try to clarify a bit, but `resolution` and `address` are orthogonal. The resolution is just about upstream host selection, the addresses is for matching incoming traffic.

Reply:
I meant to say that we should clarify how resolution DNS works with TCP..I understand address field (VIP) is about matching incoming traffic and endpoints is about where to send it if static..

Reply:
DNS resolution works the same with TCP. I am not sure why endpoints is relevant here since this is all about the routing part, probably I am missing the connection?

Reply:
May be, a part of routing is eventually being able to select endpoints for the cluster.